app/src/preHome.js

```diff
@@ -0,0 +1,201 @@
+    const [patientName, setPatientName] = useState("")
+    const [healthCareProfessionalName, setHealthCareProfessionalName] = useState("")
+    const [Sessions, setSessions] = useState([])
+
+    const history = useHistory();
+
+    const toRecordingPage = () => history.push("/recording");
+
+    const handleSubmit = event => {
+      event.preventDefault();
+      return listSessions();
+    }
+
+    const handleCreateSession = event => {
+      event.preventDefault();
+    }
+
+    async function listSessions() {
+        const apiName = 'MTADemoAPI';
+        const path = 'listSessions';
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: {
+            PatientId: searchVal,
+          },
+        };
+        const result =  await API.get(apiName, path, myInit);
+        setSessions(result.data);
+        return result;
+    }
+
+    async function listPatients() {
+        const apiName = 'MTADemoAPI';
+        const path = 'listPatients';
+        const parameters = (searchVal === '') ? {} : {PatientId: searchVal} 
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: parameters
+        };
+        const result =  await API.get(apiName, path, myInit); 
+        return result;
+    }
+
+    async function listHealthCareProfessionals() {
+        const apiName = 'MTADemoAPI';
+        const path = 'listHealthCareProfessionals';
+        const parameters = (searchVal === '') ? {} : {HealthCareProfessionalId: searchVal} 
+        const myInit = { 
+        //   headers: { 
+        //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+        //   },
+          response: true,
+          queryStringParameters: parameters
+        };
+    
+        const result =  await API.get(apiName, path, myInit); 
+        return result;
+    }
+ 
+    async function createPatient() {
+        const apiName = 'MTADemoAPI';
+        const path = 'createPatient';
+        const myInit = { 
+            //   headers: { 
+            //     Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}`,
+            //   },
+            response: true,
+            queryStringParameters: {
+                'PatientName': searchVal,
+            },
+        };
+    
+        const result =  await API.post(apiName, path, myInit); 
+        return result;
+    }
+
+    async function createHealthCareProfessional() {
+        const apiName = 'MTADemoAPI';
+        const path = 'createHealthCareProfessional';
+        const myInit = { 
+        //   headers: { 
```

Comment:
You can remove the comments if it is not needed,

Reply:
Will update it later when I solve the authorization issue.